chef/cookbooks/neutron/recipes/post_install_conf.rb

```diff
@@ -92,6 +92,49 @@ def mask_to_bits(mask)
   else
     Chef::Log.error("default provider network ml2 type driver '#{ml2_type_drivers_default_provider_network}' invalid for creating provider networks")
   end
+
+  # *START* workaround for https://bugzilla.suse.com/show_bug.cgi?id=967009
+  cookbook_file "crowbar-fix-floating-provider-attributes" do
+    source "crowbar-fix-floating-provider-attributes"
+    path "/usr/bin/crowbar-fix-floating-provider-attributes"
+    mode "0755"
+    only_if "out=$(#{neutron_cmd} net-show floating -f value -F provider:network_type); " \
+      "[ $? == 0 ] && [ ${out} != \"flat\" ]"
+  end
+
+  # This adjusts the provider attributes of the existing "floating" network
+  # to actually match what we want ("flat" network on the correct physnet)
+  execute "fix floating network provider attributes" do
+    command "/usr/bin/crowbar-fix-floating-provider-attributes " \
+      "--physnet #{ext_physnet_map["nova_floating"]} --segmentation_id 0 --type flat " \
+      "--tenant_id #{keystone_settings['service_tenant_id']} --network floating"
+    action :run
+    only_if "out=$(#{neutron_cmd} net-show floating -f value -F provider:network_type); " \
+      "[ $? == 0 ] && [ ${out} != \"flat\" ]"
+  end
+
+  file "/usr/bin/crowbar-fix-floating-provider-attributes" do
+    action :delete
+  end
```

Comment:
Yeah, because I didn't have a good only_if condition the other one would have been recreated with every chef-client run anyway. But I fixed that now with the flag on the node attributes.